### ISSUE TITLE
false type error on `Struct { ..todo!() }`

### ISSUE THREAD

**Issue by kpreid** [rust-lang/rust-analyzer#21701]
**rust-analyzer version**: 0.4.2804-standalone (05da4cf3c4 2026-02-23)

**rustc version**: rustc 1.95.0-nightly (b3869b94c 2026-02-23)

**editor or extension**: VS Code, extension pre-release 0.4.2804

**code snippet to reproduce**:
```rust
struct Struct {
    field: i32,
}

fn example() -> Struct {
    Struct { ..todo!() }
}
```

This code is accepted by rustc, but rust-analyzer says “expected Struct, found !”. It seems that the never-to-any coercion doesn’t work for the [base expression](https://doc.rust-lang.org/reference/expressions/struct-expr.html#functional-update-syntax) in particular.
